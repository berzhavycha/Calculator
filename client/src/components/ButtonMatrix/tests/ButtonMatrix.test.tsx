import React from 'react';
import { render } from '@testing-library/react';
import { ButtonMatrix } from '../index';
import { vi } from 'vitest'
import { ButtonType } from '@components';
import userEvent from '@testing-library/user-event';

describe('ButtonMatrix component', () => {
    const buttonMatrix = [
        [
            { content: '1', type: ButtonType.NUMERIC, isLastButton: false },
            { content: '+', type: ButtonType.OPERATOR, isLastButton: false },
            { content: '=', type: ButtonType.EVALUATE, isLastButton: true },
        ],
    ];

    const lastButtonRef = React.createRef<HTMLButtonElement>();
    const onButtonClick = vi.fn();

    test('should render buttons based on the provided buttonMatrix', () => {
        const { getAllByRole } = render(
            <ButtonMatrix buttonMatrix={buttonMatrix} lastButtonRef={lastButtonRef} onButtonClick={onButtonClick} />
        );

        const buttons = getAllByRole('button');
        expect(buttons.length).toBe(buttonMatrix.flat().length);

        buttonMatrix.flat().forEach((button, index) => {
            expect(buttons[index]).toHaveTextContent(button.content);
        });
    });

    test('should call onButtonClick when a button is clicked', async () => {
        const { getByText } = render(
            <ButtonMatrix buttonMatrix={buttonMatrix} lastButtonRef={lastButtonRef} onButtonClick={onButtonClick} />
        );

        const buttonContent = '1';
        const button = getByText(buttonContent);

        await userEvent.click(button);

        expect(onButtonClick).toHaveBeenCalledWith(buttonMatrix[0][0]);
    });
});
