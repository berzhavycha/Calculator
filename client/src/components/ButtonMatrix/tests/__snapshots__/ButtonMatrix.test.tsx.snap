// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ButtonMatrix component > matches the snapshot 1`] = `
<div>
  <div
    class="button-container flex flex-col items-end"
  >
    <div
      class="flex mb-4 gap-8"
    >
      <button
        class="p-2 bg-indigo-400 text-white rounded hover:bg-blue-700 h-10 w-20"
        data-calc-btn="1"
      >
        1
      </button>
      <button
        class="p-2 bg-green-400 text-white rounded hover:bg-green-500 h-10 w-20"
        data-calc-btn="+"
      >
        +
      </button>
      <button
        class="p-2 bg-purple-700 text-white rounded hover:bg-purple-800 h-10 w-20"
        data-calc-btn="="
      >
        =
      </button>
    </div>
  </div>
</div>
`;
